tenpy/networks/mpo.py

```diff
@@ -1686,7 +1696,6 @@ def travel_q_RL(i, keyL):
         if not infinite and any([ch is None for ch in charges[-1]]):
             raise ValueError("can't determine all charges on the very right leg of the MPO!")
 
-        max_checks = sys.getrecursionlimit()  # I don't expect interactions with larger range...
```

Comment:
I think it actually makes more sense to hard-code `max_checks = 1000`. All of this is only to check for errors anyways; if you have a properly set-up MPO graph (e.g. from a `CouplingMPOModel`), it will terminate after just one iteration of the loop.
You can then also remove the corresponding code in `travel_q_LR_iter` that estimates `max_checks`.